### ISSUE TITLE
Unable to Import-Excel with -Password

### ISSUE THREAD

**Issue by ca6dsm** [dfinke/ImportExcel#637]
Im unable to use the Import-Excel function with the -Password to open any xlsx files.

New-Object : Exception calling ".ctor" with "2" argument(s): "The stream must be read/write"
At C:\Program Files\WindowsPowerShell\Modules\ImportExcel\6.2.2\ImportExcel.psm1:363 char:46
+ ... elPackage = New-Object -TypeName OfficeOpenXml.ExcelPackage -Argument ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-Object], MethodInvocationException
    + FullyQualifiedErrorId : ConstructorInvokedThrowException,Microsoft.PowerShell.Commands.NewObjectCommand
 
You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\ImportExcel\6.2.2\ImportExcel.psm1:433 char:43
+             if ($Path) { $stream.close(); $ExcelPackage.Dispose() }
+                                           ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
Failed importing the Excel workbook 'I:\test\test.xlsx' with worksheet '': Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\ImportExcel\6.2.2\ImportExcel.psm1:431 char:17
+ ...     catch { throw "Failed importing the Excel workbook '$Path' with w ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Failed importin...o a null array.:String) [], RuntimeException
    + FullyQualifiedErrorId : Failed importing the Excel workbook 'I:\test\test.xlsx' with worksheet '': Cannot index into a null array.

**Comment by uSlackr** [comment]
Can you share your code? Sanitized if you need to.

**Comment by ca6dsm** [comment]
Import-Excel  -Path 'I:\test\test.xlsx'  -Password 1234

its a completely blank workbook works fine if i remove password but once i add one get the error
    On Wednesday, 24 July 2019, 02:21:45 BST, uSlackr <notifications@github.com> wrote:  
 
 
Can you share your code? Sanitized if you need to.

**Comment by uSlackr** [comment]
Maybe add some data.  I just did a quick test and it worked fine
import-excel OneDrive\Documents\Test1.xlsx -password "pass93"

\\Greg

________________________________

**Comment by ca6dsm** [comment]
added data and created on with password and one without
clear-host


import-excel -Path "C:\Test\testNoPwd.xlsx" 

import-excel -Path "C:\Test\testPwd.xlsx" -Password "pass1234"


1	2	3	4	5	6
A	A	A	A	A	A
b	b	b	b	b	b
c	c	c	c	c	c
d	d	d	d	d	d

Still getting same error.

**Comment by ca6dsm** [comment]
its failing here
 $stream = New-Object -TypeName System.IO.FileStream -ArgumentList $Path, 'Open', 'Read', 'ReadWrite'
            if ($Password) { $ExcelPackage = New-Object -TypeName OfficeOpenXml.ExcelPackage -ArgumentList $stream , $Password }
            else { $ExcelPackage = New-Object -TypeName OfficeOpenXml.ExcelPackage -ArgumentList $stream }

**Comment by jhoneill** [comment]
That looks like a regression from when I fixed the "can't read an open file" issue. 
One of the tests is meant to read and write password protected files so it should have been found. 
I'll check what's happening but it should be quite easy to fix

**Comment by jhoneill** [comment]
OK. The password test was not being run because the file ended .ps1 not .tests.ps1
It looks like the regression might be older than I thought, but I have fixed it.  It's just running its check-in tests and I'll put in a PR. 
However @dfinke  in this build of EPPlus, the crypo support for passwords is not available in PS Core 6.2
or 7 preview 2.  Do know happen to know if this addressed in later EPPlus builds ? For now I'm trapping version > 5 and not running the test.

**Comment by dfinke** [comment]
I checked a couple months back on EPPlus to see if they were further along on their port to dotnet core. Don't know about the crypto support. I didn't see much check in activity then.

**Comment by dfinke** [comment]
Fixed in v6.2.3

**Comment by pixarbuff** [comment]
Hi there - 
I'm running Import-Excel 6.2.3 and PS Core 6.2.4 on Ubuntu 18.04 and when I call 
`Import-Excel -path "./myfile.xlsx" -password "mypass"`
I get the following error:
```
Exception calling "Load" with "2" argument(s): "Could not load type 'System.Security.Cryptography.RIPEMD160Managed' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'."
At /home/ubuntu/.local/share/powershell/Modules/ImportExcel/7.0.1/Public/Import-Excel.ps1:110 char:30
+             if ($Password) { $ExcelPackage.Load($stream, $Password) }
+                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
+ FullyQualifiedErrorId : TypeLoadException
```
I see on the notes page that 

> Fix broken test & regression for passwords
>Note: Passwords do not work on pwsh. The EPPlus library does not support these dotnet core APIs at this time.

What are my options for workarounds in this situation?
